src/fetch/period_slippage.py

```diff
@@ -5,16 +5,50 @@
 from src.dune_analytics import DuneAnalytics, QueryParameter
 from src.file_io import File
 from src.models import Network
+from src.token_list import ALLOWED_TOKEN_LIST_URL, get_trusted_tokens_from_url
+
+
+def generate_sql_query_for_allowed_token_list(token_list) -> str:
+    query = "allow_listed_tokens as ( Select * from (VALUES"
+    for address in token_list:
+        query += f"('\\{address[1:]}' :: bytea),"
+    query = query[:-1]
+    query += ") AS t (token)),"
+    return query
+
+
+def adds_table_after_with_statement(query, table_to_add):
+    if query[0:4].lower() != "with":
+        raise ValueError(f"Type {query} does not start with 'with'!")
+    return "\n".join(
+        [
+            query[0:4],
+            table_to_add,
+            query[5:],
+        ]
+    )
```

Comment:
Very nice use of join here! The function name however is a bit awkward, it reads more like a mocha unit test than an actual function name. I would suggest something like `prepend_to_sub_query`